packages/utils/src/utils/split-utxos.ts

```diff
@@ -0,0 +1,36 @@
+import { ErrorCode, FuelError } from '@fuel-ts/errors';
+import type { AbstractAddress } from '@fuel-ts/interfaces';
+import type { BN } from '@fuel-ts/math';
+
+/**
+ * This function allows a consumer to split a UTXO into multiple smaller UTXOs
+ * @param balance - The total balance that the user wants to split, this is equivalent of the amount you would like to send to the recipient
+ * @param amount - The amount of each UTXO
+ * @param assetId - The asset ID of the UTXOs
+ * @param destination - The destination address for the UTXOs
+ * @param number_of_coins - The number of UTXOs to split into, it defaults to 2
+ * @returns An array of the desired UTXOs with the amount, assetId, and destination address
+ */
+export function splitUTXOs(
```

Comment:
The naming `splitUTXOs` implies that this function would be accepting actual UTXOs and creating transaction outputs that would be added to a transaction's outputs in a fashion like this:
```ts
const txRequest = new ScriptTransactionRequest();

txRequest.addCoinInput(coin1);
txRequest.addCoinInput(coin2);

const outputs = splitUTXOs({
  utxos: txRequest.getCoinInputs(),
  destination: '0x...',
  amount: bn(10),
});

txRequest.addCoinOutputs(outputs);
```

With the current implementation of this function, a more appropriate name would be `generateTransactionOutputs`. One could still use it in a similar fashion as in the code block above by summing the `amount`s of the coins, but because it's not taking in any UTXO its name is misaligned.

Reply:
@Torres-ssf @maschad I thought we had agreed about this being a docs-only thing, no?

> [!NOTE]
> Special methods in SDKs are helpful, but add a lot of complexity for documentation and future proofing

The PR isn't even merged, and we're already watching these ☝️ complexities taking shape.

Reply:
@maschad I completely agree with @arboleya here. Taking the approach of simply adding docs would make things much simpler.